### PR TITLE
Add optional seed source for seed_smf

### PR DESCRIPTION
**Pull request by citrus-it**
This allows the `seed_smf` task to take an optional `seed` parameter which defines the baseline file used for seeding. If omitted, it defaults to "global" as before.

### CODE REVIEW

**Comment by jclulow** [comment]
For my curiosity: what are you using this for?

**Comment by citrus-it** [comment]
> For my curiosity: what are you using this for?

For the image I'm working on at the moment, I want to start with something a little smaller than the global.db seed to reduce the propagation of dependencies that make it harder to untangle things later.

OmniOS kayak also currently uses `nonglobal.db` to seed the non-global ZFS image, so this is something that would be needed there if image-builder were to be used (https://github.com/omniosorg/kayak/blob/master/build/zfs_send#L169 )

File: src/main.rs [hunk 1]
```diff
@@ -2860,14 +2860,16 @@ fn run_steps(ib: &mut ImageBuilder) -> Result<()> {
                 struct SeedSmfArgs {
                     debug: Option<bool>,
                     apply_site: Option<bool>,
+                    seed: Option<String>,
                 }
 
                 let a: SeedSmfArgs = step.args()?;
                 let debug = a.debug.unwrap_or(false);
                 let apply_site = a.apply_site.unwrap_or(false);
+                let seed = a.seed.unwrap_or("global".to_string());
```
Comment (jclulow) [inline src/main.rs]:
```suggestion
                let seed = a.seed.as_deref().unwrap_or("global");
```

I think this is probably slightly preferable.  `seed` will be a `&str`, then, so you can remove the `&` on 2872.

**Review by jclulow — COMMENTED** [review]
This is a good idea.  One small nit!

File: src/main.rs (hunk 1 above)
Comment (citrus-it) [reply src/main.rs]: Done.

**Comment by citrus-it** [comment]
I've added a `skip_seed` option too which allows for seeding the SMF database starting without an initial seed. This seems to work fine for building an image.

**Pull request merged**